plugin/output/s3/s3.go

```diff
@@ -97,96 +120,97 @@ func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.OutputPluginP
 	p.logger = params.Logger
 	p.config = config.(*Config)
 
+	outPlugCount := len(p.config.MultiBuckets) + 1
+
 	// set up compression
 	newCompressor, ok := compressors[p.config.CompressionType]
 	if !ok {
 		p.logger.Fatalf("compression type: %s is not supported", p.config.CompressionType)
 	}
 	p.compressor = newCompressor(p.logger)
 
-	dir, f := filepath.Split(p.config.FileConfig.TargetFile)
+	// dir for all bucket files.
+	p.setDirs(outPlugCount)
+	// file for each bucket.
+	p.setFileNames(outPlugCount)
 
-	p.targetDir = dir
-	p.fileExtension = filepath.Ext(f)
-	p.fileName = f[0 : len(f)-len(p.fileExtension)]
-
-	p.uploadCh = make(chan string, p.config.FileConfig.WorkersCount_*4)
-	p.compressCh = make(chan string, p.config.FileConfig.WorkersCount_)
+	p.uploadCh = make(chan fileDto, p.config.FileConfig.WorkersCount_*4)
+	p.compressCh = make(chan fileDto, p.config.FileConfig.WorkersCount_)
 
 	for i := 0; i < p.config.FileConfig.WorkersCount_; i++ {
 		longpanic.Go(p.uploadWork)
 		longpanic.Go(p.compressWork)
 	}
 
-	// initialize minio client object.
-	minioClient, err := minio.New(p.config.Endpoint, p.config.AccessKey, p.config.SecretKey, p.config.Secure)
-	if err != nil || minioClient == nil {
-		p.logger.Panicf("could not create minio client, error: %s", err.Error())
-	}
-	p.client = minioClient
-
-	if p.config.client != nil {
-		p.logger.Info("set mock client")
-		p.client = *p.config.client
-	}
-
-	exist, err := p.client.BucketExists(p.config.Bucket)
-	if err != nil {
-		p.logger.Panicf("could not check bucket: %s, error: %s", p.config.Bucket, err.Error())
-	}
-	if !exist {
-		p.logger.Fatalf("bucket: %s, does not exist", p.config.Bucket)
-	}
-	p.logger.Info("client is ready")
-	p.logger.Infof("bucket: %s exists", p.config.Bucket)
-
-	anyPlugin, _ := file.Factory()
-	p.outPlugin = anyPlugin.(*file.Plugin)
+	// initialize minio clients.
+	p.clients = p.createMinioClients()
```

Comment:
Let's return error here and panic also here, not in function

Reply:
It'll help in debugging, already did it.